chcemvediet/apps/obligees/tests/__init__.py

```diff
@@ -1,25 +1,124 @@
 # vim: expandtab
 # -*- coding: utf-8 -*-
+from django.contrib.auth.models import User
 from django.template import Context, Template
 from django.test import TestCase
 
-from ..models import Obligee
+from chcemvediet.apps.geounits.models import Neighbourhood, Municipality, District, Region
+from chcemvediet.apps.inforequests.models import Inforequest, Branch, InforequestDraft
+
+from ..models import Obligee, ObligeeTag, ObligeeGroup, ObligeeAlias
+
 
 class ObligeesTestCaseMixin(TestCase):
 
-    def _create_obligee(self, omit=(), **kwargs):
-        defaults = {
+    def _pre_setup(self):
+        super(ObligeesTestCaseMixin, self)._pre_setup()
+        self.iczsj = self._create_geounit(u'00000')
+        self.user = self._create_user()
+        self.inforequest = self._create_inforequest()
+
+
+    def _call_with_defaults(self, func, kwargs, defaults):
+        omit = kwargs.pop(u'omit', [])
+        defaults.update(kwargs)
+        for key in omit:
+            del defaults[key]
+        return func(**defaults)
+
+    def _create_user(self, **kwargs):
+        try:
+            tag = u'{}'.format(User.objects.latest(u'pk').pk + 1)
+        except User.DoesNotExist:
+            tag = u'1'
+        street = kwargs.pop(u'street', u'Default User Street')
+        city = kwargs.pop(u'city', u'Default User City')
+        zip = kwargs.pop(u'zip', u'00000')
+        email_verified = kwargs.pop(u'email_verified', True)
+        user = self._call_with_defaults(User.objects.create_user, kwargs, {
+                u'username': u'default_testing_username_{}'.format(tag),
+                u'first_name': u'Default Testing First Name',
+                u'last_name': u'Default Testing Last Name',
+                u'email': u'default_testing_mail_{}@a.com'.format(tag),
+                u'password': u'default_testing_secret',
+                })
+        user.profile.street = street
+        user.profile.city = city
+        user.profile.zip = zip
+        user.profile.save()
+        if email_verified:
+            user.emailaddress_set.create(email=user.email, verified=True)
+        return user
+
+    def _create_geounit(self, id):
```

Comment:
Mam tuto metodu prepisat na 4  samostatne metody typu `call_with_defautls`, alebo to takto moze ostat?

Reply:
@viliambalaz Asi by to bolo lepsie spravit rovnako ako ostatne. Kedze v nejakych inych testoch budeme moct potrebovat namockovat priamo iba `District` napriklad.

Reply:
:ok: